pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/analytics/PlaybackSessionManager.kt

```diff
@@ -169,36 +190,37 @@ class PlaybackSessionManager {
 
     private inner class SessionManagerAnalyticsListener : PillarboxAnalyticsListener {
         override fun onPositionDiscontinuity(
-            eventTime: AnalyticsListener.EventTime,
+            eventTime: EventTime,
             oldPosition: Player.PositionInfo,
             newPosition: Player.PositionInfo,
             @DiscontinuityReason reason: Int
         ) {
             val oldItemIndex = oldPosition.mediaItemIndex
             val newItemIndex = newPosition.mediaItemIndex
 
+            if (eventTime.timeline.isEmpty) return
             DebugLogger.debug(TAG, "onPositionDiscontinuity reason = ${StringUtil.discontinuityReasonString(reason)}")
-
             if (oldItemIndex != newItemIndex && !eventTime.timeline.isEmpty) {
-                currentSession = getOrCreateSession(eventTime)
+                val newSession = getOrCreateSession(eventTime)!! // Return null only if timeline is empty
```

Comment:
Matter of personal preference, but using `checkNotNull()` would throw an `IllegalStateException` instead of a `NullPointerException`:
```suggestion
                val newSession = checkNotNull(getOrCreateSession(eventTime)) // Return null only if timeline is empty
```